src/program.rs

```diff
@@ -0,0 +1,66 @@
+use starknet_crypto::{poseidon_hash_many, FieldElement};
+use std::collections::HashMap;
+use std::convert::TryInto;
+
+use crate::parse_raw;
+
+pub struct ExtractProgramResult {
+    pub program: Vec<FieldElement>,
+    pub program_hash: FieldElement,
+}
+
+pub fn extract_program(input: String) -> anyhow::Result<ExtractProgramResult> {
+    // Parse the input string into a proof structure
+    let proof = parse_raw(&input)?;
+
+    // Retrieve the program segment from the proof
+    let program_segment = proof
+        .public_input
+        .segments
+        .first()
```

Comment:
`first` is more idiomatic I agree, but if the offsets of the different segments are well known, it would be easier for the reader to have constants definition.

```rust
const PROGRAM_SEGMENT_OFFSET = 0;
const EXECUTION_SEGMENT_OFFSET = 1;
//...
```

Reply:
fixed